client.go

```diff
@@ -451,3 +454,108 @@ func (c *Client) Version(ctx context.Context) (Version, error) {
 		Revision: response.Revision,
 	}, nil
 }
+
+type imageFormat string
+
+const (
+	ociImageFormat imageFormat = "oci"
+)
+
+type importOpts struct {
+	path string
+}
+
+type ImportOpt func(c *importOpts) error
+
+// TODO: add support for tar writer?
+type exportOpts struct {
+	format              imageFormat
+	path                string
+	translateMediaTypes bool
+}
+
+type ExportOpt func(c *exportOpts) error
+
+// WithOCIDirectoryExportation returns ExportOpt for local OCI directory.
+// Path must be a valid path to non-existing or existing OCI directory.
+// If the directory exists, the exported image will be added to the existing image index.
+func WithOCIDirectoryExportation(path string) ExportOpt {
+	return func(c *exportOpts) error {
+		c.format = ociImageFormat
+		c.path = path
+		return nil
+	}
+}
+
+// TODO: add WithOCITarExportation?
+// TODO: add WithRawDirectoryExportation maybe? (raw rootfs directory)
+
+// WithMediaTypeTranslation transforms media types according to the format.
+// e.g. application/vnd.docker.image.rootfs.diff.tar.gzip
+//      -> application/vnd.oci.image.layer.v1.tar+gzip
+func WithMediaTypeTranslation(b bool) ExportOpt {
+	return func(c *exportOpts) error {
+		c.translateMediaTypes = b
+		return nil
+	}
+}
+
+func (c *Client) Import(ctx context.Context, ref string, opts ...ImportOpt) (Image, error) {
+	return nil, errors.New("client.Import is not implemented")
+}
+
+func (c *Client) Export(ctx context.Context, desc ocispec.Descriptor, opts ...ExportOpt) error {
```

Comment:
Instead of taking a path, would it be better to have export work over a writer? 

`Export() (io.ReadCloser, error)`?

This way, clients have more flexibility for how they handle the exported content, they can push it over a stream or write it to disk at their path.  Right now this client is not very flexible when you take a path like this